sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/degrade/DegradeSlot.java

```diff
@@ -15,30 +15,73 @@
  */
 package com.alibaba.csp.sentinel.slots.block.degrade;
 
+import java.util.List;
+
+import com.alibaba.csp.sentinel.Entry;
 import com.alibaba.csp.sentinel.context.Context;
 import com.alibaba.csp.sentinel.node.DefaultNode;
 import com.alibaba.csp.sentinel.slotchain.AbstractLinkedProcessorSlot;
 import com.alibaba.csp.sentinel.slotchain.ProcessorSlot;
 import com.alibaba.csp.sentinel.slotchain.ResourceWrapper;
+import com.alibaba.csp.sentinel.slots.block.BlockException;
+import com.alibaba.csp.sentinel.slots.block.degrade.circuitbreaker.CircuitBreaker;
 import com.alibaba.csp.sentinel.spi.SpiOrder;
+import com.alibaba.csp.sentinel.util.TimeUtil;
 
 /**
- * A {@link ProcessorSlot} dedicates to {@link DegradeRule} checking.
+ * A {@link ProcessorSlot} dedicates to circuit breaking.
  *
- * @author leyou
+ * @author Carpenter Lee
+ * @author Eric Zhao
  */
 @SpiOrder(-1000)
 public class DegradeSlot extends AbstractLinkedProcessorSlot<DefaultNode> {
 
     @Override
-    public void entry(Context context, ResourceWrapper resourceWrapper, DefaultNode node, int count, boolean prioritized, Object... args)
-        throws Throwable {
-        DegradeRuleManager.checkDegrade(resourceWrapper, context, node, count);
+    public void entry(Context context, ResourceWrapper resourceWrapper, DefaultNode node, int count,
+                      boolean prioritized, Object... args) throws Throwable {
+        performChecking(resourceWrapper);
+
         fireEntry(context, resourceWrapper, node, count, prioritized, args);
     }
 
+    void performChecking(ResourceWrapper r) throws BlockException {
+        List<CircuitBreaker> circuitBreakers = DegradeRuleManager.getCircuitBreakers(r.getName());
+        if (circuitBreakers == null || circuitBreakers.isEmpty()) {
+            return;
+        }
+        for (CircuitBreaker cb : circuitBreakers) {
+            if (!cb.tryPass()) {
+                throw new DegradeException(cb.getRule().getLimitApp(), cb.getRule());
+            }
+        }
+    }
+
     @Override
-    public void exit(Context context, ResourceWrapper resourceWrapper, int count, Object... args) {
-        fireExit(context, resourceWrapper, count, args);
+    public void exit(Context context, ResourceWrapper r, int count, Object... args) {
+        Entry curEntry = context.getCurEntry();
+        if (curEntry.getBlockError() != null) {
+            fireExit(context, r, count, args);
+            return;
+        }
+        List<CircuitBreaker> circuitBreakers = DegradeRuleManager.getCircuitBreakers(r.getName());
+        if (circuitBreakers == null || circuitBreakers.isEmpty()) {
+            fireExit(context, r, count, args);
+            return;
+        }
+
+        if (curEntry.getBlockError() == null) {
```

Comment:
This judgement seems can be removed since Line#63.

Reply:
It's here in case the blockError was modified.